pygenesys/utils/tsprocess.py

```diff
@@ -198,152 +194,178 @@ def weekly_hourly(data_path, N_seasons=52, N_hours=24, kind='demand'):
         The number of seasons in the energy system model.
     N_hours : integer
         The hourly resolution of the energy system model.
+    how : string
+        The time series aggregation method. Only used in ``tsprocess.create_timeslices``
+        which depends on the ``tsam`` package.
 
     Returns
     -------
     distribution : numpy array
         The time series data distributed over the specified time
         slices.
     """
-    try:
-        time_series = pd.read_csv(data_path,
+
+    if isinstance(dataframe, str):
+        time_series = pd.read_csv(dataframe,
                                   usecols=[0, 1],
                                   index_col=['time'],
                                   parse_dates=True,
                                   )
-    except BaseException:
-        except_string = """
-                        Could not import time series data. Check the following:
+    elif isinstance(dataframe, pd.DataFrame):
+        time_series = dataframe
 
-                        1. That the path to the file is correct.
-                        2. The dataset has two columns: "time" and a column
-                           with your data.
-                        3. The "time" column has a pandas datetime index.
-                        """
-        print(except_string)
+    aggregation = tsam.TimeSeriesAggregation(time_series,
+                                             noTypicalPeriods=n_seasons,
+                                             hoursPerPeriod=24,
+                                             clusterMethod=how)
 
-    time_series = time_series.resample('H').mean()
-    time_series.interpolate('linear', inplace=True)
-    years_grouped = time_series.groupby(time_series.index.year)
-    data_list = []
-    for year in years_grouped.groups:
-        data = years_grouped.get_group(year)
-        if len(data) >= 8760:
-            data_list.append(np.array(data.iloc[:, 0])[:8760])
-        else:
-            pass
-            # data_list.append(np.array(data.iloc[:, 0]))
-    data_list = np.array(data_list)
-    average_profile = data_list.mean(axis=0)
-    if kind.lower() == "demand":
-        daily_hourly_profile = average_profile / average_profile.sum()
-    elif kind.lower() == "cf":
-        daily_hourly_profile = (
-            average_profile / (time_series.iloc[:, 0].max()))
-    return daily_hourly_profile
+    typPeriods = aggregation.createTypicalPeriods()
+
+    if normalize == 'demand':
+        sum_val = typPeriods.iloc[:, 0].sum()
+        typPeriods.iloc[:, 0] = typPeriods.iloc[:, 0] / sum_val
+
+    elif normalize == 'cf':
+        max_val = typPeriods.iloc[:, 0].max()
+        typPeriods.iloc[:, 0] = typPeriods.iloc[:, 0] / max_val
+
+    profile = typPeriods.iloc[:, 0].values
+
+    return profile
 
 
 if __name__ == '__main__':
 
     from pygenesys.data.library import campus_stm_demand, campus_elc_demand
     from pygenesys.data.library import railsplitter_data, solarfarm_data
```

Comment:
Are these dependencies specific to your work? If so, it might be better to just have this import matplotlib and not these data.

Reply:
The ``if __name__ == '__main__':`` block is not run unless the file itself is sourced. It's meant to store an example, but the conditional block cannot be empty.